components/plugin_marketplace/marketplace_item/marketplace_item.test.js

```diff
@@ -4,93 +4,308 @@
+            installedVersion: '0.0.1',
+            isInstalling: false,
+            onUpdate: () => {},
+        };
 
-    test('should render with no plugin icon', () => {
-        const props = {...baseProps};
-        delete props.iconData;
+        describe('should render nothing', () => {
+            it('when no installed version', () => {
+                const props = {
+                    ...baseProps,
+                    installedVersion: '',
+                };
+                const wrapper = mount(
+                    <UpdateDetails {...props}/>
+                );
 
-        const wrapper = shallow(
-            <MarketplaceItem {...props}/>
-        );
+                expect(wrapper.isEmptyRender()).toBe(true);
+            });
 
-        expect(wrapper).toMatchSnapshot();
-    });
+            it('when installed version matches available version', () => {
+                const props = {
+                    ...baseProps,
+                    installedVersion: baseProps.availableVersion,
+                };
+                const wrapper = mount(
+                    <UpdateDetails {...props}/>
+                );
 
-    test('should render with no homepage url', () => {
-        const props = {...baseProps};
-        delete props.homepageUrl;
+                expect(wrapper.isEmptyRender()).toBe(true);
+            });
 
-        const wrapper = shallow(
-            <MarketplaceItem {...props}/>
-        );
+            it('when installed version is newer than available version', () => {
+                const props = {
+                    ...baseProps,
+                    installedVersion: '0.0.3',
+                };
+                const wrapper = mount(
+                    <UpdateDetails {...props}/>
+                );
 
-        expect(wrapper).toMatchSnapshot();
-    });
+                expect(wrapper.isEmptyRender()).toBe(true);
+            });
 
-    test('should render with server error', () => {
-        const props = {
-            ...baseProps,
-            error: 'An error occurred.',
-        };
+            it('when installing', () => {
+                const props = {
+                    ...baseProps,
+                    isInstalling: true,
+                };
+                const wrapper = mount(
+                    <UpdateDetails {...props}/>
+                );
 
-        const wrapper = shallow(
-            <MarketplaceItem {...props}/>
-        );
+                expect(wrapper.isEmptyRender()).toBe(true);
+            });
+        });
 
-        expect(wrapper).toMatchSnapshot();
+        it('should render without release notes url', () => {
+            const props = {
+                ...baseProps,
+                releaseNotesUrl: '',
+            };
+
+            const wrapper = mount(
+                <UpdateDetails {...props}/>
+            );
+
+            expect(wrapper).toMatchSnapshot();
+        });
+
+        it('should render without release notes url', () => {
```

Comment:
Should this be `with release notes url`?
```suggestion
        it('should render with release notes url', () => {
```